backend/TraderX/src/main/java/cmpe451/group6/rest/portfolio/service/PortfolioService.java

```diff
@@ -289,4 +289,24 @@ public PortfolioResponseDTO getPortfolio(String username, String portfolioName)
 
     }
 
+    /**
+     * Returns a list consisting of names of portfolios of "requester"
+     * 
+     * @param requesterName
+     * @return List of Strings
+     */
+    public List<String> getSelfPortfolios(String requesterName) {
+
+        List<String> portfolioNames = new ArrayList<String>();
+
+
+        for (Portfolio portfolio : portfolioRepository.findByUser_username(requesterName)) {
```

Comment:
Modifying the query in the repository such that ‘SELECT p.name’ would do the same work as this for loop does. IMO fetching all portfolio objects is not reasonable and causes extra load on the service when we need only names of them.